### DISCUSSION TITLE
How to create an evil text object for bash functions using emacs-tree-sitter?

### DISCUSSION THREAD

**Discussion by NightMachinery** [emacs-tree-sitter/elisp-tree-sitter#159] [Q&A]
How to create an evil text object for bash functions using emacs-tree-sitter?

**Comment by theHamsta** [comment]
I have no idea on how to do that in emacs. But it would be great to team up and share queries as we have already a bunch of textobjects defined here for Neovim https://github.com/nvim-treesitter/nvim-treesitter-textobjects. In case anyone wants to start a similar project for emacs

**Comment by ubolonton** [comment]
This depends on the mechanism to define new text objects in `evil`. `tree-sitter` helps with locating the objects.

For a simple text object like a function, `tree-sitter-node-at-point` could be enough.

A more generic implementation could define interfaces to plug in language-specific text-object queries, like what @theHamsta mentioned above.

**Comment by meain** [comment]
I have started a project which does this. You can check it out at https://github.com/meain/evil-textobj-tree-sitter . The queries are actually coming from the neovim project like @theHamsta mentioned.
